sky/templates/kubernetes-ray.yml.j2

```diff
@@ -309,14 +312,17 @@ available_node_types:
         {% endif %}
 
         # Add node selector if GPU/TPUs are requested:
-        {% if (k8s_topology_label_key is not none and k8s_topology_label_value is not none) or (k8s_spot_label_key is not none) %}
+        {% if (k8s_topology_label_key is not none and k8s_topology_label_value is not none) or (k8s_spot_label_key is not none) or (k8s_enable_flex_start) %}
         nodeSelector:
             {% if k8s_topology_label_key is not none and k8s_topology_label_value is not none %}
             {{k8s_topology_label_key}}: {{k8s_topology_label_value}}
             {% endif %}
             {% if k8s_spot_label_key is not none %}
             {{k8s_spot_label_key}}: {{k8s_spot_label_value|tojson}}
             {% endif %}
+            {% if k8s_enable_flex_start %}
```

Comment:
should we change it to `enable_dws` then, considering we will always enable flex-start with dws?

Reply:
We still need to distinguish between the two modes internally, because the nodeSelector item here is only added for the `flex_start` mode and not necessary for `flex_start_queued_provisioning` mode.